### PR TITLE
bngblaster: init at 0.8.28

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 268174` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>bngblaster</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libdict</li>
  </ul>
</details>

```
       > CMake Error at CMakeLists.txt:10 (message):
       >   Host Darwin is not supported!
```

**Comment by yu-re-ka** [comment]
Updated to 0.8.29

**Comment by vidister** [comment]
@yu-re-ka  with the new release we could remove the patches, right?

**Comment by yu-re-ka** [comment]
Patches removed. tested build on x86_64-linux

File: pkgs/by-name/bn/bngblaster/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  cmakeFlags = [
+    "-DBNGBLASTER_TESTS=${if finalAttrs.doCheck then "ON" else "OFF"}"
+    "-DBNGBLASTER_VERSION=${finalAttrs.version}"
+  ];
+
+  doCheck = true;
+
+  meta = with lib; {
+    homepage = "https://github.com/rtbrick/bngblaster/";
+    changelog = "https://github.com/rtbrick/bngblaster/releases/tag/${version}";
```
Comment (Janik-Haag) [inline pkgs/by-name/bn/bngblaster/package.nix:47]: doesn't this need to be:
```suggestion
    changelog = "https://github.com/rtbrick/bngblaster/releases/tag/${finalAttrs.version}";
```

Comment (yu-re-ka) [reply pkgs/by-name/bn/bngblaster/package.nix]: Indeed. Thanks

**Review by Janik-Haag — APPROVED** [review]

**Pull request merged**